python/lsst/ts/m2gui/utility_monitor.py

```diff
@@ -881,3 +911,71 @@ def update_position(
                 position[idx] = round(component, self.NUM_DIGIT_AFTER_DECIMAL)
 
             signal.emit(position)
+
+    def update_net_force_moment_total(
+        self,
+        component_x: float,
+        component_y: float,
+        component_z: float,
+        is_force: bool = True,
+    ) -> None:
+        """Update the total net force or moment.
+
+        Parameters
+        ----------
+        component_x : `float`
+            X component. Unit is Newton or Newton * meter based on 'is_force'.
+        component_y : `float`
+            Y component. Unit is Newton or Newton * meter based on 'is_force'.
+        component_z : `float`
+            Z component. Unit is Newton or Newton * meter based on 'is_force'.
+        is_force : `bool`, optional
+            Is the force data or not. If not, the moment data is applied. (the
+            default is True)
+        """
+
+        # Decide the value to compare and the related signal
+        value_new = [component_x, component_y, component_z]
+        value_old = self.net_force_total if is_force else self.net_moment_total
+
+        signal = (
+            self.signal_net_force_moment.net_force_total
+            if is_force
+            else self.signal_net_force_moment.net_moment_total
+        )
+
+        tol = get_tol(self.NUM_DIGIT_AFTER_DECIMAL)
+        if self._has_changed(np.array(value_old), np.array(value_new), tol):
+            for idx, value in enumerate(value_new):
+                value_old[idx] = round(value, self.NUM_DIGIT_AFTER_DECIMAL)
+
+            signal.emit(value_old)
+
+    def update_force_balance(
+        self, fx: float, fy: float, fz: float, mx: float, my: float, mz: float
+    ) -> None:
+        """Update the force balance status.
```

Comment:
This can benefit from the same. Just pass vector in, using again list comprehension to build it:

[message["f" + axis] for axis in "xyz"] + message["m" + axis] for axis in "xyz"]

Reply:
Fixed in the commit: [e5a46e3](https://github.com/lsst-ts/ts_m2gui/pull/45/commits/e5a46e3e907479c4ce0e51b07878165a73b2cad5) Thanks!